### PR TITLE
Newsletter: move email settings to its own section

### LINKED ISSUE #79822: Move "Newsletter" settings to its own section
Project thread peKye1-iw-p2

Currently Newsletter settings are under **Settings → Reading**.

_Newsletter_ doesn’t really fit inside _Reading_ or _Writing_, because settings include enabling new features (Subscribe Modal), customizing welcome emails, etc.

Separate sections also makes it clearer for customer that they have a "blog" and a "newsletter" (one comes with another at WP.com).

image

image

### PR DESCRIPTION
**Pull request by simison**
Resolves https://github.com/Automattic/wp-calypso/issues/79822

Adding menu item in a separate PR: https://github.com/Automattic/jetpack/pull/32146

## Proposed Changes

* Moves Newsletter settings to their own section

image

image

I'm simply moving  the settings as-is, but in a follow-up we can re-organise the settings a bit. E.g. split into specific "email" and "site" related settings.

## Testing Instructions

- If you want to test sidebar navigation, try together with https://github.com/Automattic/jetpack/pull/32146
- There's a feature flag to adjust few things, but the settings page is always accessible at `/settings/newsletter/SITE`
- The section at "Reading" goes away with feature flag enabled: `/settings/reading/SITE?flags=settings/newsletter-settings-page`
- The task at newsletter launchpad at my home points to right page depending on feature flag

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Comment by simison** [comment]
Needs a rebase.

**Comment by simison** [comment]
@crisbusquets, I added a simple "moved" notification to old placement. Let us know if you have thoughts on how it should look like/copy. We can leave it there for ~2 weeks at least and then clean it out.

image

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8716862

Thank you @simison for including a screenshot in the description! This is really helpful for our translators.

File: client/my-sites/site-settings/settings-newsletter/main.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+};
+
+type Fields = {
+	subscription_options?: SubscriptionOptions;
+	wpcom_featured_image_in_email?: boolean;
+	wpcom_subscription_emails_use_excerpt?: boolean;
+	sm_enabled?: boolean;
+};
+
+const getFormSettings = ( settings: unknown & Fields ) => {
```
Comment (mashikag) [inline client/my-sites/site-settings/settings-newsletter/main.tsx]: Why do we have the union with `unknown` type here? Also, one line below, it seems like you are doing a check for `! settings` which implies to me that the argument is `optional`.

File: client/my-sites/site-settings/settings-newsletter/main.tsx [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+	return {
+		...( subscription_options && { subscription_options } ),
+		wpcom_featured_image_in_email: !! wpcom_featured_image_in_email,
+		wpcom_subscription_emails_use_excerpt: !! wpcom_subscription_emails_use_excerpt,
+		sm_enabled: !! sm_enabled,
+	};
+};
+
+const connectComponent = connect( ( state: IAppState ) => {
```
Comment (mashikag) [inline client/my-sites/site-settings/settings-newsletter/main.tsx]: I may be wrong, but it looks to me as if we are not using any of the connected props. 🤔

**Review by mashikag — APPROVED** [review]
The code works as expected and is clear to read. 👌 

I just noticed 2 small issues (see my inline comments), but I would not consider them blocking the PR.

File: client/my-sites/site-settings/settings-newsletter/main.tsx (hunk 1 above)
Comment (simison) [reply client/my-sites/site-settings/settings-newsletter/main.tsx]: Agree it's weird.

I'm simply moving the code around so this came from existing Reading settings file where these settings originated:

https://github.com/Automattic/wp-calypso/blob/e74e4d189869af331d56d9419dde6ea5ac500c5b/client/my-sites/site-settings/settings-reading/main.tsx#L48-L51

File: client/my-sites/site-settings/settings-newsletter/main.tsx (hunk 2 above)
Comment (simison) [reply client/my-sites/site-settings/settings-newsletter/main.tsx]: Good catch! This also came from Reading settings I think — a good for a clean-up afterwards, or in this PR.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8716862

Thank you @simison for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8716862).

**Pull request merged**